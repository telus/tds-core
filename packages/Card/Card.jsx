import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Box from '@tds/core-box'
import Image from '@tds/core-image'
import { responsiveProps } from '@tds/util-prop-types'

import { colorWhite, colorWhiteLilac, colorGreyAthens, colorGreyGainsboro } from '@tds/core-colours'
import { borders } from '@tds/shared-styles'
import { safeRest, handleResponsiveStyles } from '@tds/util-helpers'

import { deprecate } from '../../shared/utils/warn'

const getVariant = ({ variant }) => {
  if (variant === 'white' || variant === 'default' || variant === 'defaultWithBorder') {
    return {
      boxShadow: '0 0 16px 0 rgba(0, 0, 0, 0.1)',
      backgroundColor: colorWhite,
      border: variant === 'defaultWithBorder' ? `1px solid ${colorGreyGainsboro}` : undefined,
    }
  }
  if (variant === 'lavender' || variant === 'branded') {
    return {
      backgroundColor: colorWhiteLilac,
    }
  }
  return { backgroundColor: colorGreyAthens }
}

const deprecationWarning = deprecatedVariant => {
  const variants = {
    white: 'default',
    lavendar: 'branded',
    grey: 'alternative',
  }
  return `The ${deprecatedVariant} variant has been deprecated. Please use the '${
    variants[deprecatedVariant]
  }' variant.`
}

export const StyledCard = styled(({ fullHeight, fullBleedImage, ...props }) => <Box {...props} />)(
  borders.none,
  borders.rounded,
  getVariant,
  ({ fullHeight }) => {
    if (fullHeight) {
      return { height: '100%' }
    }
    return {}
  }
)

const ContainerStyles = fullBleedImage =>
  fullBleedImage &&
  fullBleedImage.position &&
  handleResponsiveStyles({ position: fullBleedImage.position }, ({ position }) => {
    if (!fullBleedImage) return {}
    const direction = {
      left: 'row',
      right: 'row-reverse',
      top: 'column',
      bottom: 'column-reverse',
      none: 'row',
    }
    const styles = {
      display: 'flex',
      flexDirection: direction[position],
      justifyContent: 'space-between',
      '> img': {
        display: position === 'none' ? 'none' : 'block',
        margin: 'auto',
      },
    }
    return styles
  })

const StyledDiv = styled.div(ContainerStyles)

/**
 * A content container.
 *
 * @version ./package.json
 */
const Card = ({ variant, children, fullHeight, spacing, fullBleedImage, ...rest }) => {
  if (variant === 'white' || variant === 'lavendar' || variant === 'grey') {
    deprecate('@tds/core-card', deprecationWarning(variant))
  }

  const spacingProps = {}
  if (spacing === 'default') {
    spacingProps.vertical = 5
    spacingProps.horizontal = 4
  } else if (spacing === 'narrow') {
    spacingProps.vertical = 4
    spacingProps.horizontal = 3
  } else if (spacing === 'compact') {
    spacingProps.inset = 3
  } else if (spacing === 'intermediate') {
    spacingProps.inset = 4
  }

  return (
    <StyledCard {...safeRest(rest)} fullHeight={fullHeight} variant={variant}>
      {fullBleedImage ? (
        <StyledDiv {...fullBleedImage}>
          <Image
            src={fullBleedImage.src}
            width={fullBleedImage.width}
            height={fullBleedImage.height}
            alt={fullBleedImage.alt}
          />
          <Box {...spacingProps}>{children}</Box>
        </StyledDiv>
      ) : (
        <Box {...spacingProps}>{children}</Box>
      )}
    </StyledCard>
  )
}

Card.propTypes = {
  /**
   * The style.
   *
   * @since 2.1.0  added `default`, `defaultWithBorder`, `branded`, `alternative`.
   *
   * **Deprecated:** `white`, `lavendar`,`grey`
   */
  variant: PropTypes.oneOf([
    'white',
    'lavender',
    'grey',
    'default',
    'branded',
    'alternative',
    'defaultWithBorder',
  ]),
  /**
   * The content. Can be text, any HTML element, or any component.
   */
  children: PropTypes.node.isRequired,
  /**
   * Sets the `Card`'s `height` equal to its parent.
   */
  fullHeight: PropTypes.bool,
  spacing: PropTypes.oneOf(['default', 'narrow', 'compact', 'intermediate']),
  fullBleedImage: PropTypes.shape({
    src: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    alt: PropTypes.string.isRequired,
    position: responsiveProps(PropTypes.oneOf(['left', 'right', 'top', 'bottom', 'none']))
      .isRequired,
  }),
}

Card.defaultProps = {
  variant: 'default',
  fullHeight: false,
  spacing: 'default',
  fullBleedImage: undefined,
}

export default Card
