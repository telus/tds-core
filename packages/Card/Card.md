Cards will stretch to fill the width of their container. Wrap the card in another element to apply size.

```jsx
<Card>
  <Box between={4}>
    <Box between={3}>
      <Heading level="h2">Need a hand?</Heading>
      <Paragraph>
        Ready to order? Have a question? We'll get back to you, with volume discounts available to
        larger accounts.
      </Paragraph>
    </Box>

    <div>
      {/* We wrap ButtonLink in a div to prevent flex styling from applying to it */}
      <ButtonLink href="#">Request sales callback</ButtonLink>
    </div>
  </Box>
</Card>
```

### Coloured variants

Card also supports an 'alternative' or 'branded' `variant`. These appear flat, while the default Card appears floating.

```jsx
<Box inline between={3}>
  <Card variant="branded">
    <Box between={3}>
      <Heading level="h4">Holiday deal</Heading>

      <Paragraph size="medium">
        Get a new smartphone for $0
        <br />
        on a 2-year plan.
      </Paragraph>
    </Box>
  </Card>

  <Card variant="alternative">
    <Box between={3}>
      <Heading level="h4">Holiday deal</Heading>

      <Paragraph size="medium">
        Get a new smartphone for $0
        <br />
        on a 2-year plan.
      </Paragraph>
    </Box>
  </Card>

  <Card>
    <Box between={3}>
      <Heading level="h4">Find the right gift</Heading>

      <Paragraph>
        Find something they'll love from our
        <br />
        selection of great devices.
      </Paragraph>

      <div>
        <ChevronLink variant="primary" href="#">
          Explore latest devices
        </ChevronLink>
      </div>
    </Box>
  </Card>
</Box>
```

### Cards with borders

Teams can use the `defaultWithBorder` variant to add a border to the `Card` for comparison, horizontal plan, and My TELUS account overview cards.

```jsx
<FlexGrid>
  <FlexGrid.Row>
    <FlexGrid.Col xs={12} md={5}>
      <Card variant="defaultWithBorder">
        <Box between={3}>
          <Heading level="h3">Internet 15 - Special Offer</Heading>
          <PriceLockup
            size="medium"
            price="55"
            rateText="per month"
            bottomText="for 24 months, then $75 per month"
            signDirection="left"
          />
          <HairlineDivider />
          <Text size="medium">Good for basic browsing, and posting to social media.</Text>
          <BenefitWithHeading>
            <BenefitWithHeading.Item icon={Speed} heading="15 mbps">
              download speed
            </BenefitWithHeading.Item>
            <BenefitWithHeading.Item icon={Speed} heading="1 mbps">
              upload speed
            </BenefitWithHeading.Item>
            <BenefitWithHeading.Item icon={DataLimit} heading="Unlimited">
              Monthly data
            </BenefitWithHeading.Item>
          </BenefitWithHeading>
          <Notification variant="branded">
            <Text bold>Offer:</Text> Order Internet online on a 2 year term and get a $150 bill
            credit.
          </Notification>
          <Button>Add to cart</Button>
          <ChevronLink href="#">Learn more</ChevronLink>
        </Box>
      </Card>
    </FlexGrid.Col>
  </FlexGrid.Row>
</FlexGrid>
```

### Full Height Cards

In some cases you want cards to match the height of their parent so that the bottom edge of the cards are aligned. This is usually needed when using cards in a `FlexGrid`. Use the `fullHeight` prop to achieve this.

```jsx
<FlexGrid>
  <FlexGrid.Row>
    <FlexGrid.Col xs={12} md={4}>
      <Card fullHeight>
        <Box between={4}>
          <Heading level="h3">Reward yourself</Heading>
          <Text>
            When you bundle TELUS SmartHome Security with one other eligible TELUS service, you
            qualify for TELUS Rewards. Earn points just by paying your monthly bill, and then redeem
            great rewards, like gift cards, contests, TELUS bill credits, and the latest tech.
          </Text>
        </Box>
      </Card>
    </FlexGrid.Col>
    <FlexGrid.Col xs={12} md={4}>
      <Card fullHeight>
        <Box between={4}>
          <Heading level="h3">Enjoy fast internet in every corner of your home</Heading>
          <Text>
            TELUS Boost Wi-Fi creates a mesh network throughout your home so your devices will
            always connect to the strongest possible Wi-Fi.
          </Text>
        </Box>
      </Card>
    </FlexGrid.Col>
    <FlexGrid.Col xs={12} md={4}>
      <Card fullHeight>
        <Box between={4}>
          <Heading level="h3">
            Power your home security and home monitoring system with the fastest internet technology
          </Heading>
          <Text>
            TELUS PureFibre is number one internet technology for speed and reliability. Save $10/
            month on TELUS SmartHome Security when you bundle with TELUS Internet.
          </Text>
        </Box>
      </Card>
    </FlexGrid.Col>
  </FlexGrid.Row>
</FlexGrid>
```

### Card Padding

By _Default_, cards have Box 5 spacing on top/bottom and Box 4 on left/right. They are used for marketing pages to call out or break information.

<img src="card/card-padding-default.png" alt="card padding default" style="max-width: 100%" />

_Narrow_ cards have Box 4 spacing on top/bottom and Box 3 on left/right. This card is used when presenting plans to a customer.

<img src="card/card-padding-narrow.png" alt="card padding narrow" style="max-width: 100%" />

**Cards with even padding around them are used in internal applications or pages to help with categorizing items and de-emphasize in terms of visual hierarchy.**

_Compact_ cards have Box 3 padding all around it.

<img src="card/card-padding-compact.png" alt="card padding compact" style="max-width: 100%" />

_Intermediate_ cards have Box 4 padding all around it.

<img src="card/card-padding-intermediate.png" alt="card padding intermediate" style="max-width: 100%" />

### Full Bleed Image Card

In cases where designs should have images without margins (full bleed images), use the Full Bleed Image property. If passing a full bleed image, the src, width, height, alt tag, and position are required.

Position can be passed as one of the following words: 'left', 'right', 'top', 'bottom', or 'none', depending on where in the card you would like the image to be. This property may also be passed as a responsive prop, using the TDS breakpoints. A full bleed image with position: {xs: 'none', md: 'left'} for example, will have a full bleed image to the left of card content when viewed on desktops, and no image when viewed on mobile screens.

In a horizontal card the image height should be appropriately matched to that of the card.
In a vertical card, the width of the image would ideally match the width of the card. In cases where the image is smaller than the width of the card, the full bleed image will be centered in the card.

#### Horizontal card with a full bleed image positioned right on desktop, none on mobile

```jsx
<Card
  fullBleedImage={{
    src: 'card/salal_branch_tall.png',
    width: 191,
    height: 176,
    alt: 'salal branch image',
    position: { xs: 'none', md: 'right' },
  }}
  variant="defaultWithBorder"
>
  <Box between={2}>
    <Heading level="h3">Your first bill isn't ready yet.</Heading>
    <Paragraph>
      We'll send you an email or text message when it's available. Want to pay your bills
      automatically?
      <ChevronLink href="#" variant="secondary">
        Set up pre-authorized payments
      </ChevronLink>
    </Paragraph>
  </Box>
</Card>
```

#### Vertical card with a full bleed image positioned bottom on both desktop and mobile

```jsx
<FlexGrid>
  <FlexGrid.Row>
    <FlexGrid.Col xs={12} md={5}>
      <Card
        fullBleedImage={{
          src: 'card/salal_branch.png',
          width: 398,
          height: 147,
          alt: 'salal branch image',
          position: 'bottom',
        }}
        variant="defaultWithBorder"
      >
        <Box between={2}>
          <Heading level="h3">Your first bill isn't ready yet.</Heading>
          <Paragraph>
            We'll send you an email or text message when it's available. Want to pay your bills
            automatically?
            <ChevronLink href="#" variant="secondary">
              Set up pre-authorized payments
            </ChevronLink>
          </Paragraph>
        </Box>
      </Card>
    </FlexGrid.Col>
  </FlexGrid.Row>
</FlexGrid>
```
