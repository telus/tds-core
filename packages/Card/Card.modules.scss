@import '../colour/colours';

.card {
  composes: none rounded from '../../shared/styles/Borders.modules.scss';
}

.white {
  composes: card;

  box-shadow: 0 0 16px 0 rgba(213, 213, 213, 0.5);
  background-color: $color-white;
}

.lavender {
  composes: card;

  background-color: $color-white-lilac;
}

.grey {
  composes: card;

  background-color: $color-athens-grey;
}
