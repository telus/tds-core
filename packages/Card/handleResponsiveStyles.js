import { prepareArray, generateStyles } from '@tds/util-helpers'

const MOBILE_BREAKPOINTS = ['xs', 'sm']
const DESKTOP_BREAKPOINTS = ['md', 'lg', 'xl']

export const isMobileBreakpoint = breakpoint => MOBILE_BREAKPOINTS.indexOf(breakpoint) !== -1
export const isDesktopBreakpoint = breakpoint => DESKTOP_BREAKPOINTS.indexOf(breakpoint) !== -1

const handleBoundaryCrossing = (acc, curr) => {
  if (
    isMobileBreakpoint(curr.from) &&
    ((curr.until !== 'md' && isDesktopBreakpoint(curr.until)) || typeof curr.until === 'undefined')
  ) {
    const props = Object.keys(curr.props).filter(
      prop => typeof curr.props[prop] === 'number' && curr.props[prop] > 3
    )
    if (props.length !== 0) {
      const mobileBreakpoint = { ...curr, props: curr.props }
      const desktopBreakpoint = { ...curr, props: curr.props }

      mobileBreakpoint.until = 'md'
      desktopBreakpoint.from = 'md'

      return acc.concat([mobileBreakpoint, desktopBreakpoint])
    }
  }
  return acc.concat([curr])
}

const handleResponsiveStyles = (props, styleFn) => {
  const breakpoints = prepareArray(props)
    .filter(bp => Object.keys(bp.props).length > 0)
    .reduce(handleBoundaryCrossing, [])
  return generateStyles(breakpoints, styleFn)
}

export default handleResponsiveStyles
