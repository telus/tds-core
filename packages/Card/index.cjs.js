const Card = require('./dist/index.cjs')

module.exports = Card
