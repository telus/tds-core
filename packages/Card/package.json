{
  "name": "@tds/core-card",
  "version": "2.3.9",
  "description": "A content container",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/core-box": "^2.3.0",
    "@tds/core-colours": "^2.2.1",
    "@tds/shared-styles": "^2.0.0",
    "@tds/core-image": "^2.0.20",
    "@tds/core-responsive": "^1.3.3",
    "@tds/util-helpers": "^1.4.2",
    "prop-types": "^15.5.10",
    "@tds/util-prop-types": "^1.4.0"
  }
}
