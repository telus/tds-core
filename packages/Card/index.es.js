import Card from './dist/index.es'

export default Card
