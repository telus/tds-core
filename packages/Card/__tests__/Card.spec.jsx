import React from 'react'
import { shallow, mount, render } from 'enzyme'

import Card from '../Card'

describe('<Card />', () => {
  const doShallow = (props = {}) => shallow(<Card {...props}>Some content</Card>)
  const doMount = (props = {}) => mount(<Card {...props}>Some content</Card>)

  it('renders', () => {
    const card = render(<Card>Children</Card>)

    expect(card).toMatchSnapshot()
  })

  it('can be presented as one of the allowed variants', () => {
    let card = doMount()
    expect(card).toMatchSnapshot()

    card = doMount({ variant: 'default' })
    expect(card).toMatchSnapshot()

    card = doMount({ variant: 'branded' })
    expect(card).toMatchSnapshot()

    card = doMount({ variant: 'alternative' })
    expect(card).toMatchSnapshot()

    card = doMount({ variant: 'defaultWithBorder' })
    expect(card).toMatchSnapshot()
  })

  it('renders with full height', () => {
    const card = mount(<Card fullHeight>Children</Card>)

    expect(card).toHaveStyleRule('height', '100%')
  })

  it('passes additional attributes to the input element', () => {
    const card = doShallow({ role: 'some-role', 'data-some-value': 'some value' })

    expect(card).toHaveProp('role', 'some-role')
    expect(card).toHaveProp('data-some-value', 'some value')
  })

  it('does not allow custom CSS', () => {
    const card = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(card).not.toHaveProp('className', 'my-custom-class')
    expect(card).not.toHaveProp('style')
  })

  describe('spacing', () => {
    it('renders default spacing', () => {
      const card = doMount({ spacing: 'default' })
      expect(card).toMatchSnapshot()
    })

    it('renders narrow spacing', () => {
      const card = doMount({ spacing: 'narrow' })
      expect(card).toMatchSnapshot()
    })

    it('renders compact spacing', () => {
      const card = doMount({ spacing: 'compact' })
      expect(card).toMatchSnapshot()
    })

    it('renders intermediate spacing', () => {
      const card = doMount({ spacing: 'intermediate' })
      expect(card).toMatchSnapshot()
    })
  })

  describe('full bleed image', () => {
    it('renders without an image', () => {
      const card = doMount()
      expect(card).toMatchSnapshot()
    })

    it('renders with an image', () => {
      const card = doMount({
        fullBleedImage: {
          src: 'image-example.jpg',
          width: 200,
          height: 200,
          alt: 'image',
          position: 'top',
        },
      })
      expect(card).toMatchSnapshot()
    })

    it('renders with responsive image props', () => {
      const card = doMount({
        fullBleedImage: {
          src: 'image-example.jpg',
          width: 200,
          height: 200,
          alt: 'image',
          position: { xs: 'top', md: 'bottom' },
        },
      })
      expect(card).toMatchSnapshot()
    })
  })
})
