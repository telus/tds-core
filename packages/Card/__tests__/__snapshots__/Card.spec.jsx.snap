// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> can be presented as one of the allowed variants 1`] = `
.c1 {
  display: block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 2rem;
  padding-bottom: 2rem;
}

.c0 {
  border-width: 0;
  border-radius: 4px;
  box-shadow: 0 0 16px 0 rgba(213,213,213,0.5);
  background-color: #fff;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 2rem;
    padding-right: 2rem;
    padding-top: 3rem;
    padding-bottom: 3rem;
  }
}

<Card
  variant="white"
>
  <Styled(Component)
    horizontal={4}
    variant="white"
    vertical={5}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      horizontal={4}
      variant="white"
      vertical={5}
    >
      <Component
        className="c0"
        horizontal={4}
        variant="white"
        vertical={5}
      >
        <Box
          className="c0"
          horizontal={4}
          inline={false}
          tag="div"
          vertical={5}
        >
          <styled.div
            className="c0"
            horizontal={4}
            inline={false}
            tag="div"
            vertical={5}
          >
            <StyledComponent
              className="c0"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    [Function],
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      [Function],
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              horizontal={4}
              inline={false}
              tag="div"
              vertical={5}
            >
              <div
                className="c0 sc-bwzfXH c0 c1"
              >
                Some content
              </div>
            </StyledComponent>
          </styled.div>
        </Box>
      </Component>
    </StyledComponent>
  </Styled(Component)>
</Card>
`;

exports[`<Card /> can be presented as one of the allowed variants 2`] = `
.c1 {
  display: block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 2rem;
  padding-bottom: 2rem;
}

.c0 {
  border-width: 0;
  border-radius: 4px;
  box-shadow: 0 0 16px 0 rgba(213,213,213,0.5);
  background-color: #fff;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 2rem;
    padding-right: 2rem;
    padding-top: 3rem;
    padding-bottom: 3rem;
  }
}

<Card
  variant="white"
>
  <Styled(Component)
    horizontal={4}
    variant="white"
    vertical={5}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      horizontal={4}
      variant="white"
      vertical={5}
    >
      <Component
        className="c0"
        horizontal={4}
        variant="white"
        vertical={5}
      >
        <Box
          className="c0"
          horizontal={4}
          inline={false}
          tag="div"
          vertical={5}
        >
          <styled.div
            className="c0"
            horizontal={4}
            inline={false}
            tag="div"
            vertical={5}
          >
            <StyledComponent
              className="c0"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    [Function],
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      [Function],
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              horizontal={4}
              inline={false}
              tag="div"
              vertical={5}
            >
              <div
                className="c0 sc-bwzfXH c0 c1"
              >
                Some content
              </div>
            </StyledComponent>
          </styled.div>
        </Box>
      </Component>
    </StyledComponent>
  </Styled(Component)>
</Card>
`;

exports[`<Card /> can be presented as one of the allowed variants 3`] = `
.c1 {
  display: block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 2rem;
  padding-bottom: 2rem;
}

.c0 {
  border-width: 0;
  border-radius: 4px;
  background-color: #f2eff4;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 2rem;
    padding-right: 2rem;
    padding-top: 3rem;
    padding-bottom: 3rem;
  }
}

<Card
  variant="lavender"
>
  <Styled(Component)
    horizontal={4}
    variant="lavender"
    vertical={5}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      horizontal={4}
      variant="lavender"
      vertical={5}
    >
      <Component
        className="c0"
        horizontal={4}
        variant="lavender"
        vertical={5}
      >
        <Box
          className="c0"
          horizontal={4}
          inline={false}
          tag="div"
          vertical={5}
        >
          <styled.div
            className="c0"
            horizontal={4}
            inline={false}
            tag="div"
            vertical={5}
          >
            <StyledComponent
              className="c0"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    [Function],
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      [Function],
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              horizontal={4}
              inline={false}
              tag="div"
              vertical={5}
            >
              <div
                className="c0 sc-bwzfXH c0 c1"
              >
                Some content
              </div>
            </StyledComponent>
          </styled.div>
        </Box>
      </Component>
    </StyledComponent>
  </Styled(Component)>
</Card>
`;

exports[`<Card /> can be presented as one of the allowed variants 4`] = `
.c1 {
  display: block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 2rem;
  padding-bottom: 2rem;
}

.c0 {
  border-width: 0;
  border-radius: 4px;
  background-color: #f7f7f8;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 2rem;
    padding-right: 2rem;
    padding-top: 3rem;
    padding-bottom: 3rem;
  }
}

<Card
  variant="grey"
>
  <Styled(Component)
    horizontal={4}
    variant="grey"
    vertical={5}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      horizontal={4}
      variant="grey"
      vertical={5}
    >
      <Component
        className="c0"
        horizontal={4}
        variant="grey"
        vertical={5}
      >
        <Box
          className="c0"
          horizontal={4}
          inline={false}
          tag="div"
          vertical={5}
        >
          <styled.div
            className="c0"
            horizontal={4}
            inline={false}
            tag="div"
            vertical={5}
          >
            <StyledComponent
              className="c0"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    [Function],
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      [Function],
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              horizontal={4}
              inline={false}
              tag="div"
              vertical={5}
            >
              <div
                className="c0 sc-bwzfXH c0 c1"
              >
                Some content
              </div>
            </StyledComponent>
          </styled.div>
        </Box>
      </Component>
    </StyledComponent>
  </Styled(Component)>
</Card>
`;

exports[`<Card /> renders 1`] = `
.c1 {
  display: block;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  padding-top: 2rem;
  padding-bottom: 2rem;
}

.c0 {
  border-width: 0;
  border-radius: 4px;
  box-shadow: 0 0 16px 0 rgba(213,213,213,0.5);
  background-color: #fff;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 2rem;
    padding-right: 2rem;
    padding-top: 3rem;
    padding-bottom: 3rem;
  }
}

<div
  class="c0 sc-bwzfXH c0 c1"
>
  Children
</div>
`;
